### ISSUE TITLE
AI catapults should fire at troops as well

### ISSUE THREAD

**Issue by PitchNeeded** [Sh0wdown/UnofficialCrusaderPatch#217]
Catapults can be very successful when they fire at lots of troops together, for example siege forces gathering to besiege a castle.

It would be good if AI catapults could fire at big groups of enemy troops as well as just buildings. This might be hard to impliment though, as the groups of enemy troops would have to be big enough to be effective targets, it wouldn't be good if AI catapults fired at individual enemy units, as they would be unlikely to hit the units.

If it was implimented, it would open the option of putting 'defensive catapults' on AIVS too to fire stones at approaching slow melee troops, or to throw cows at them if the AI had dairy farms.

**Comment by PitchNeeded** [comment]
Also can we have better cow throwing from Wolf, Pig and Sheriff's catapults? They throw cows up against enemy walls which often isn't very effective.

Although cows should be changed to affect troops on the tops of walls or towers if the cow lands there.

**Comment by Monsterfisch** [comment]
ah I was wrong about the ammunition then but regardless of that I don't think that ai would be well adviced to have catapults shoot at units considering they might simply end up getting caught up shooting people instead of walls since they are not very accurate unlike balistas which already take much longer to get close to the enemy castles.

**Comment by PitchNeeded** [comment]
> ah I was wrong about the ammunition then but regardless of that I don't think that ai would be well adviced to have catapults shoot at units considering they might simply end up getting caught up shooting people instead of walls since they are not very accurate unlike balistas which already take much longer to get close to the enemy castles.

Well, ideally, they would only shoot at big enough groups of units, like enemy siege forces grouped together before they advance. It could be an option for non-siege catapults that many of the AI lords create (the harassment catapults)' before they move within firing range of enemy walls.

**Comment by Monsterfisch** [comment]
right I mean if there is a way to test for group sizes like that I could see it being cool behavior. just not sure how feasible it is in the bigger picture considering we have attack troops standing around idly atm getting annihilated by mangonels.. essentially i am saying make the attacks stronger so defense tactics like this eve make sense otherwise defending ai vs ai is too "easy"

**Comment by PitchNeeded** [comment]
Yeah the AI's melee troops wait around too long before advancing I think, this makes them vulnerable to mangonels and tower ballistae fire

**Comment by Monsterfisch** [comment]
same thing with troops gathering up in enemy fire before storming the castle

**Comment by PitchNeeded** [comment]
> right I mean if there is a way to test for group sizes like that I could see it being cool behavior. just not sure how feasible it is in the bigger picture considering we have attack troops standing around idly atm getting annihilated by mangonels.. essentially i am saying make the attacks stronger so defense tactics like this eve make sense otherwise defending ai vs ai is too "easy"

Yes that could be a problem, it could potentially make AI attacks less effective against each other. But I still think it makes sense for AIs to use catapults more cleverly.

Most AIs do use catapults so at least it would be quite balanced. Although an AI like Wazir or Nizar might suffer as they don't use catapults, but you could always put defensive catapults in their AIVs if this proposed feature was enabled.
